### PR TITLE
tests: add tests for verification

### PR DESCRIPTION
**Pull request by bnjbvr**
Part of #2602.

### CODE REVIEW

File: crates/matrix-sdk/tests/integration/encryption.rs [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+};
+use serde_json::{json, value::RawValue};
+use wiremock::{
+    matchers::{method, path},
+    Mock, MockServer, Request, ResponseTemplate,
+};
+
+type DeviceKeys = Arc<Mutex<BTreeMap<OwnedUserId, BTreeMap<String, Box<RawValue>>>>>;
+
+struct MockedServer {
```
Comment (poljar) [inline crates/matrix-sdk/tests/integration/encryption.rs:29]: Eyyy, nice.

File: crates/matrix-sdk/tests/integration/encryption.rs [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+    fn add_known_device(&mut self, device_id: &DeviceId) {
+        self.known_devices.lock().unwrap().insert(device_id.to_string());
+    }
+}
+
+async fn bootstrap_cross_signing_and_verify_device_test_helper(client: &Client) {
+    client.encryption().bootstrap_cross_signing(None).await.unwrap();
+
+    let status = client.encryption().cross_signing_status().await.unwrap();
+    assert!(status.has_master);
```
Comment (poljar) [inline crates/matrix-sdk/tests/integration/encryption.rs:121]: Perhaps you'll want this from the secret storage PR: https://github.com/matrix-org/matrix-rust-sdk/pull/2621/files#diff-5ef8b27df74094f85e75c840dff614573eb9bf72f03663622facfcfb9aaba68cR112-R117

File: crates/matrix-sdk/tests/integration/encryption.rs [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+    assert!(alice_verifies_bob_request.is_none());
+
+    let bob_device = alice.encryption().get_device(&bob_user_id, &bob_device_id).await.unwrap();
+    assert!(bob_device.is_none());
+
+    // Have both Alice and Bob bootstrap cross-signing.
+    bootstrap_cross_signing_and_verify_device_test_helper(&alice).await;
+    bootstrap_cross_signing_and_verify_device_test_helper(&bob).await;
+
+    // TODO: continue here with mutual verification using sas
```
Comment (poljar) [inline crates/matrix-sdk/tests/integration/encryption.rs:236]: Yeah that one will be tricky. Perhaps you as a first step you can manually verify the other persons user identity and check that the device is verified as well. That one only requires the signatures to be put into a `/keys/query` after correctly calling `UserIdentity::verify()`.

File: crates/matrix-sdk/tests/integration/encryption.rs [hunk 4]
```diff
@@ -0,0 +1,237 @@
+use std::{
```
Comment (poljar) [inline crates/matrix-sdk/tests/integration/encryption.rs:1]: Can you put ~all of this code into a `verification` submodule? The secret storage PR already adds a `secrets_storage.rs` submodule: https://github.com/matrix-org/matrix-rust-sdk/pull/2621/files#diff-cc6e326da953cebbe55173940c6bda458f880aaabbf06e5bbd869f62f1afb1b8R1-R3

Should make the merge conflict easier to handle.

**Review by poljar — APPROVED** [review]
Left some nits, looks good and thanks for tackling this.

File: crates/matrix-sdk/tests/integration/encryption.rs (hunk 2 above)
Comment (poljar) [reply crates/matrix-sdk/tests/integration/encryption.rs:121]: Actually, that's now part of https://github.com/matrix-org/matrix-rust-sdk/pull/2758/.

File: crates/matrix-sdk/tests/integration/encryption.rs (hunk 4 above)
Comment (bnjbvr) [reply crates/matrix-sdk/tests/integration/encryption.rs:1]: Sure!

File: crates/matrix-sdk/tests/integration/encryption.rs (hunk 3 above)
Comment (bnjbvr) [reply crates/matrix-sdk/tests/integration/encryption.rs]: Yeah, I've removed that second test since it's quite incomplete; will add in a followup.

**Pull request merged**